forms/nht/grantsAndOthersProgressReport.json

```diff
@@ -16856,6 +16856,809 @@
+                "defaultValue": 0,
+                "behaviour": [
+                  {
+                    "condition": "participationIndividualsOrGroups == \"Individuals\"",
+                    "type": "enable"
+                  }
+                ],
+                "validate": [
+                  {
+                    "rule": "required"
+                  },
+                  {
+                    "rule": "min[0]"
+                  },
+                  {
+                    "param": {
+                      "expression": "participationIndividualNumber",
+                      "type": "computed"
+                    },
+                    "rule": "max"
+                  }
+                ]
+              },
+              {
+                "dataType": "text",
+                "name": "firstNationGroupNumber",
+                "behaviour": [
+                  {
+                    "condition": "participationIndividualsOrGroups == \"Groups\"",
+                    "type": "enable"
+                  }
+                ],
+                "constraints": [
+                  "0",
+                  "1-10",
+                  "11-20",
+                  "21-30",
+                  "31-40",
+                  "41-50"
+                ],
+                "validate": [
+                  {
+                    "rule": "required"
+                  },
+                  {
+                    "param": {
+                      "expression": "participationGroupNumber",
+                      "type": "computed"
+                    },
+                    "rule": "max",
+                    "message": "The number of first nation participants cannot be greater than the total number of participants"
+                  }
+                ]
+              },
+              {
+                "dataType": "text",
+                "name": "participationDetails",
+                "validate": "required, maxSize[3000]"
+              }
+            ],
+            "dataType": "list",
+            "name": "participationInformationTableIfm"
+          },
+          {
+            "columns": [
+              {
+                "dataType": "text",
+                "name": "initialOrFollowup",
+                "constraints": [
+                  "Initial",
+                  "Follow-up"
+                ],
+                "validate": "required"
+              },
+              {
+                "dataType": "number",
+                "name": "numberImagesCollected",
+                "validate": "required"
+              },
+              {
+                "dataType": "text",
+                "name": "imageFilename",
+                "description": "All image files must be named using the following naming convention: Image pairing_Intial/follow up_Site name_Image #_Date (image taken)_Progress Report #/ Final Report_Project ID Example: Image pairing_Intial_Site 1_Image 1_08052025_Progress report 1_URCP000777",
```

Comment:
If there is a forced naming convention for the filename the validator supports a regular expression rule you could probably enforce that.

Reply:
@chrisala 
So far the only user feedback is the addition of time field, but I can discuss/open this to her. 
Can you please share if we have a report that already using reg expression validation?